Source/SteveSloth/Private/MyGenericEnemyPatrolState.cpp

```diff
@@ -14,16 +14,24 @@ void UMyGenericEnemyPatrolState::EnterState()
 {
 	Player = USteveSingleton::GetSteve()->GetPlayerCharacter();
 	Steve = Cast<AMyPlayer>(Player);
+
+	StartingSpot = Myself->GetStartingLocation().GetLocation();
+	PatrolSpot = StartingSpot + UKismetMathLibrary::RandomUnitVector() * FMath::FRandRange(0.f, Myself->GetPatrolRange());
+
 	IsAnimationRunning = false;
 }
 
 void UMyGenericEnemyPatrolState::ExitState()
 {
-
+	
 }
 
 void UMyGenericEnemyPatrolState::UpdateState(float deltaTime)
 {
+	FVector currentSpot = Myself->GetStartingLocation().GetLocation();
+	FVector locationToGo = FMath::VInterpConstantTo(currentSpot, PatrolSpot, deltaTime, 100.f);
```

Comment:
should the 100f based on move speed? (not 100% about this one)

Reply:
this one is based off of getting delta time to 1 instead of 1000